client/verta/verta/endpoint/_endpoint.py

```diff
@@ -213,17 +213,32 @@ def _update_from_build(self, update_body, wait=False):
             print("waiting for update...", end='')
             sys.stdout.flush()
 
-            status_dict = self.get_status()  # if you think of a better var name, please do use it haha
-            while status_dict['status'] not in ("active", "error") \
-                    or (status_dict['status'] == "active" and len(self.get_status()['components']) > 1):
+            build_status = self._get_build_status(update_body['build_id'])
+            # have to check using build status, otherwise might never terminate
+            while build_status['status'] not in ("finished", "error"):
+                print(".", end='')
+                sys.stdout.flush()
+                time.sleep(5)
+                build_status = self._get_build_status(update_body['build_id'])
+
+            if build_status["status"] == "error":
+                print()
+                failure_msg = build_status.get('message', "no error message available")
+                raise RuntimeError("endpoint update failed;\n{}".format(failure_msg))
+
+            # still need to wait because status might be "creating" even though build status is "finished"
+            status_dict = self.get_status()
+            while status_dict["status"] not in ("active", "error") or \
+                    (status_dict['status'] == "active" and len(status_dict['components']) > 1):
                 print(".", end='')
                 sys.stdout.flush()
                 time.sleep(5)
                 status_dict = self.get_status()
 
             print()
-            if self.get_status()['status'] == "error":
-                failure_msg = self.get_status()['components'][0].get('message', "no error message available")
+            # is this check necessary?
```

Comment:
Why wouldn't it be?

Reply:
A wrong assumption I used to have. Removed.